lib/AST/GenericSignature.cpp

```diff
@@ -422,7 +422,8 @@ getSubstitutionMap(TypeSubstitutionFn subs,
     Type currentReplacement = depTy.subst(subs, lookupConformance,
                                           SubstFlags::UseErrorType);
     if (auto paramTy = dyn_cast<GenericTypeParamType>(canTy))
-      subMap.addSubstitution(paramTy, currentReplacement);
+      if (!currentReplacement->hasError())
```

Comment:
`compiler_crashers_fixed/01762-swift-constraints-constraintsystem-opengeneric.swift`, among many others, hits this path.